sql/core_clients_daily_v1.sql

```diff
@@ -1,23 +1,52 @@
-CREATE TEMP FUNCTION udf_mode_last(x ANY TYPE) AS ((
-  SELECT
-    val
-  FROM (
+CREATE TEMP FUNCTION
+  udf_json_mode_last(list ANY TYPE) AS ((
+    SELECT
+      ANY_VALUE(_value)
+    FROM
+      UNNEST(list) AS _value
+    WITH
+    OFFSET
+      AS _offset
+    GROUP BY
+      TO_JSON_STRING(_value)
+    ORDER BY
+      COUNT(_value) DESC,
+      MAX(_offset) DESC
+    LIMIT
+      1));
+  --
+CREATE TEMP FUNCTION
+  udf_mode_last(list ANY TYPE) AS ((
     SELECT
-      val,
-      COUNT(val) AS n,
-      MAX(offset) AS max_offset
+      _value
     FROM
-      UNNEST(x) AS val
-    WITH OFFSET AS offset
+      UNNEST(list) AS _value
+    WITH
+    OFFSET
+      AS
+    _offset
     GROUP BY
-      val
+      _value
     ORDER BY
-      n DESC,
-      max_offset DESC
-  )
-  LIMIT 1
-));
-
+      COUNT(_value) DESC,
+      MAX(_offset) DESC
```

Comment:
No way. I refuse to believe you can put aggregations directly in the order by.

Reply:
I was just fretting earlier in the day about whether we could possibly remove a subquery from this UDF, and couldn't come up with anything. This is exciting.